tool/speed.cc

```diff
@@ -1029,8 +1029,25 @@ static bool SpeedECDSACurve(const std::string &name, int nid,
   }
 
   BM_NAMESPACE::UniquePtr<EC_KEY> key(EC_KEY_new_by_curve_name(nid));
-  if (!key ||
-      !EC_KEY_generate_key(key.get())) {
+  if(!key) {
+    return false;
+  }
+
+  TimeResults results;
+  if (!TimeFunction(&results, [&key]() -> bool {
+// We run extra checks on the key in FIPS mode
+#if defined(BORINGSSL_FIPS)
+        return EC_KEY_generate_key_fips(key.get()) == 1;
+#else
+        return EC_KEY_generate_key(key.get()) == 1;
+#endif
+      })) {
+    return false;
+  }
+
+  results.Print(name + " key generation");
+
+  if (!EC_KEY_generate_key(key.get())) {
     return false;
   }
```

Comment:
is this check this make redundant by the null check + return on L1045?

Reply:
Kinda? I'd say there's not really a reason we should expect this to fail if our key generation benchmark succeeded, but it doesn't hurt and if this failed we'd probably want the benchmarking run to fail.

Reply:
sure, it's not inside a timed block so i suppose it doesn't hurt.